mmf/trainers/core/training_loop.py

```diff
@@ -199,6 +197,12 @@ def _finish_update(self):
                 self.config,
                 scale=self.scaler.get_scale(),
             )
+        if getattr(self.config.training, 'device', 'cuda') == 'xla' and self.config.distributed.world_size > 1: 
```

Comment:
Can you use the `is_xla` util function here? Also, no need for checking the world size. reduce_gradients is a noop if wsz is 1.